tests/compatibility-test.py

```diff
@@ -438,6 +456,50 @@ def get_new_value():
         client.close()
 
 
+class RayServiceTestCase(unittest.TestCase):
+    service_config_file = 'tests/config/ray-service.yaml'
+    service_serve_update_config_file = 'tests/config/ray-service-serve-update.yaml'
+    service_cluster_update_config_file = 'tests/config/ray-service-cluster-update.yaml'
+
+    @classmethod
+    def setUpClass(cls):
+        # Ray cluster is running inside a local Kind environment.
+        # We use port mapping to connect to the Kind environment
+        # from another local ray container. The local ray container
+        # outside Kind environment has the same ray version as the
+        # ray cluster running inside Kind environment.
+        delete_cluster()
+        create_cluster()
+        apply_kuberay_resources()
+        download_images()
+        create_kuberay_service(RayServiceTestCase.service_config_file)
+        cls.port_forwarding_proc = subprocess.Popen('kubectl port-forward service/rayservice-sample-serve-svc 8000', shell=True)
+
+    def test_ray_serve_work(self):
+        curl_cmd = 'curl  -X POST -H \'Content-Type: application/json\' localhost:8000 -d \'["MANGO", 2]\''
+        wait_for_condition(
+            lambda: shell_run(curl_cmd) == 0,
+            timeout=5,
+        )
+        create_kuberay_service(RayServiceTestCase.service_serve_update_config_file)
+        curl_cmd = 'curl  -X POST -H \'Content-Type: application/json\' localhost:8000 -d \'["MANGO", 2]\''
+        time.sleep(5)
+        wait_for_condition(
+            lambda: shell_run(curl_cmd) == 0,
+            timeout=60,
+        )
+        create_kuberay_service(RayServiceTestCase.service_cluster_update_config_file)
+        curl_cmd = 'curl  -X POST -H \'Content-Type: application/json\' localhost:8000 -d \'["MANGO", 2]\''
```

Comment:
I'm not really following the serve examples, but
it seems here we're posting a mango and just checking that the mango was posted successfully.

Are there some results of some sort we could check for?

Reply:
Yes. In general, we can check the result.
It should be 6 first. And then after updating, it should 8.
But I did not figure out a good way to do it in python for now.
So this pr just checks the call succeeds.

Reply:
ok, sounds ok for now